test/Microsoft.NET.TestFramework/Commands/TestCommand.cs

```diff
@@ -170,5 +171,29 @@ public static void LogCommandResult(ITestOutputHelper log, CommandResult result)
                 log.WriteLine($"Exit Code: {result.ExitCode}");
             }
         }
+
+        public static string GetPreviousDotnetVersion()
+        {
+            string currentFramework = ToolsetInfo.CurrentTargetFramework;
+            var match = Regex.Match(currentFramework, @"net(\d+)\.(\d+)");
```

Comment:
```suggestion
            var match = Regex.Match(currentFramework, @"^net(\d+)\.(\d+)$");
```

It looks like you are assuming the value will start with net, and end with 2 parts only. TFM identifiers like net9.0.0 or net6.0-windows are valid, but would be mangled by the logic below, so it might be better to be explicit about the expectations

Reply:
Thanks for the remark:)